backend/controllers/gameController.js

```diff
@@ -32,331 +35,458 @@ const QUESTIONS_PER_GAME = 10;
 let lastRun = Date.now();
 
 
-/*
-  Checks that paramId is valid id number: 24 chars, contains only: 0-9, a-f and A-F characters
-*/
-const checkValidityOfId = (paramId) => {
-
-  if ( !paramId )
-    return false;
-
-  //Check if parameter paramId is in valid format
-  if (!paramId.match(/^[0-9a-fA-F]{24}$/))
-    return false;
-
-  return true;
-};
-
-/*
-  Creates array which hold n ammount of random entries got from param object.
-*/
-const getRandomQuestions = (param, n) => {
-  let questionIdArray = [];
-
-  //store all the question ids to array
-  for (let question of param) {
-    questionIdArray.push(question.id);
-  }
-
-  // Shuffle array
-  let shuffled = questionIdArray.sort(() => 0.5 - Math.random());
-
-  // Get sub-array of first n elements after shuffled
-  return shuffled.slice(0, n);
-};
-
 
 /* 
   Get game by id
 */
-exports.get_game = async function (req, res) {
+exports.get_game = [
+  check('id', 'Invalid game id or game does not exist')
+    .exists()
+    .isMongoId()
+    .custom((val) => Game.isValidGame(val))
+    .bail(),
```

Comment:
Unnecessary bail, doesn't have any effect if used at the end of the validation chain

Reply:
fixed em